plugins/catalog-backend/src/ingestion/processors/ldap/client.ts

```diff
@@ -31,8 +32,20 @@ import {
 export class LdapClient {
   private vendor: Promise<LdapVendor> | undefined;
 
-  static async create(target: string, bind?: BindConfig): Promise<LdapClient> {
+  static async create(
+    logger: Logger,
+    target: string,
+    bind?: BindConfig,
+  ): Promise<LdapClient> {
     const client = ldap.createClient({ url: target });
+
+    // We want to have a catch-all error handler at the top, since the default
+    // behavior of the client is to blow up the entire process when it fails,
+    // unless an error handler is set.
+    client.on('error', (err: ldap.Error) => {
+      logger.warn(`LDAP client threw an error, ${errorString(err)}`);
```

Comment:
Prolly better to log this with error level right?

Reply:
K!